presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4

```diff
@@ -238,7 +238,7 @@ valueExpression
 primaryExpression
     : NULL                                                                           #nullLiteral
     | interval                                                                       #intervalLiteral
-    | identifier STRING                                                              #typeConstructor
+    | type STRING                                                                    #typeConstructor
```

Comment:
Isn't this out of scope of this pull request? Can we create an issue for "disallow arbitrary types in type constructor"? Especially that @losipiuk said that such solution was discussed with you about constructing decimal literals.